src/strategies/moca-staking/index.ts

```diff
@@ -0,0 +1,80 @@
+import { formatUnits } from '@ethersproject/units';
+import { multicall } from '../../utils';
+
+export const author = 'mocaverse';
+export const version = '0.1.0';
+
+const abi = [
+  {
+    inputs: [
+      {
+        internalType: 'address',
+        name: 'user',
+        type: 'address'
+      }
+    ],
+    name: 'getUser',
+    outputs: [
+      {
+        components: [
+          {
+            internalType: 'uint256',
+            name: 'amount',
+            type: 'uint256'
+          },
+          {
+            internalType: 'uint256',
+            name: 'cumulativeWeight',
+            type: 'uint256'
+          },
+          {
+            internalType: 'uint256',
+            name: 'lastUpdateTimestamp',
+            type: 'uint256'
+          }
+        ],
+        internalType: 'struct SimpleStaking.Data',
+        name: '',
+        type: 'tuple'
+      }
+    ],
+    stateMutability: 'view',
+    type: 'function'
+  }
+];
+
+function getArgs(options, address: string) {
+  const args: Array<string | number> = options.args || ['%{address}'];
+  return args.map((arg) =>
+    typeof arg === 'string' ? arg.replace(/%{address}/g, address) : arg
+  );
+}
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+  const response = await multicall(
+    network,
+    provider,
+    abi,
+    addresses.map((address: any) => [
+      options.address,
+      'getUser',
+      getArgs(options, address)
+    ]),
+    { blockTag }
+  );
+
+  return Object.fromEntries(
+    response.map(([value], i) => [
```

Comment:
@ChaituVR I have tried contract call on both local test and testnet playground by creating a space but seems not able to get the voting power.

I believe because of the `[value]` here

Reply:
ah correct!